cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -384,6 +384,31 @@ public ApplicationMeta getLatest(NamespaceId namespaceId, String appName) throws
     return null;
   }
 
+  public Collection<ApplicationId> getLatestAppIds(NamespaceId namespace, Collection<String> appNames)
+    throws IOException {
+    Range range = Range.singleton(
+      ImmutableList.of(
+        Fields.stringField(StoreDefinition.AppMetadataStore.NAMESPACE_FIELD, namespace.getNamespace())));
+    // scan based on: latest field set to true
+    Field<?> latestField = Fields.stringField(StoreDefinition.AppMetadataStore.LATEST_FIELD, "true");
+    Set<String> appSet = new HashSet<>(appNames);
+
+    List<ApplicationId> appIds = new ArrayList<>();
+    // TODO: switch to multiScan after CDAP-19781 is done
```

Comment:
@chtyim can use multiScan(`["namepsace", "appName", latest="true"]`) once https://github.com/cdapio/cdap/pull/14647 is merged